From 0718bc489244d2b822552b7b7edcb48d29df173e Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Fri, 21 Apr 2023 15:12:48 +0500
Subject: [PATCH] (halium) do not exit if unable to find logical partition

This is non-fatal as mounts can be handled outside LXC container.

Change-Id: I46d0b3908a44017bebf64c8624f507422d682eaa
---
 main.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/main.cpp b/main.cpp
index 5b1bdf7..abb3a68 100644
--- a/main.cpp
+++ b/main.cpp
@@ -251,7 +251,10 @@ static int process_config(VolumeManager* vm, VoldConfigs* configs) {
         /* Make sure logical partitions have an updated blk_device. */
         if (entry.fs_mgr_flags.logical && !fs_mgr_update_logical_partition(&entry) &&
             !entry.fs_mgr_flags.no_fail) {
-            PLOG(FATAL) << "could not find logical partition " << entry.blk_device;
+            //PLOG(FATAL) << "could not find logical partition " << entry.blk_device;
+            // Halium: make it non-fatal
+            PLOG(ERROR) << "could not find logical partition " << entry.blk_device;
+            continue;
         }
 
         if (entry.mount_point == "/data" && !entry.metadata_key_dir.empty()) {
-- 
2.39.2

